src/shared/dissect-image.c

```diff
@@ -3909,7 +3909,8 @@ int verity_dissect_and_mount(
                 const char *required_host_os_release_id,
                 const char *required_host_os_release_version_id,
                 const char *required_host_os_release_sysext_level,
-                const char *required_sysext_scope) {
+                const char *required_sysext_scope,
+                DissectedImage **ret_image) {
```

Comment:
this makes sense for me to add (i.e. both the new return param, and making `dest` optional)

but i wouldn't call this "2 steps". I mean, you can use it like that, but the comment below is a bit confusing, since it suggests that was the only way really.